### ISSUE TITLE
Xmrig appeared to be "killed" and huge pages allocation is 0%

### ISSUE THREAD

**Issue by ghost** [xmrig/xmrig#2649]
![unknown-37.png](https://user-images.githubusercontent.com/12180913/138814715-2fe40e42-8e58-4ac5-89e5-9378c1b6026d.png)

**Comment by DeeDeeRanged** [comment]
Have you enabled msr in grub? Maybe thats why msr and hugepages, need to run as root, are not working.

**Comment by Footsiefat** [comment]
Yeah i dont think i have. Is there any way to run it without changing grub? Just curious :)
@DeeDeeRanged

**Comment by DeeDeeRanged** [comment]
@Footsiefat 
In /etc/default grub add change the follwing:
GRUB_CMDLINE_LINUX_DEFAULT="quiet"
GRUB_CMDLINE_LINUX="msr.allow_writes=on"

run sudo update-grub.

MSR has been default disable in the kernel on linux.
Which distro are you running as gcc-5 is ancient by now.

**Comment by Footsiefat** [comment]
Its a vps so i probably cant acess grub thats why :/ @DeeDeeRanged

**Comment by DeeDeeRanged** [comment]
@Footsiefat 
Thats probably also the reason you cannot get hugepages. AFAIK msr and hugepages do not work with vps.

**Comment by Footsiefat** [comment]
@DeeDeeRanged  Thanks for the info! Just out of curiousity any there any way to make this work without having hugepages enabled?

**Comment by Spudz76** [comment]
I would imagine it's killing anything that allocated 3GB of any page size of memory.  Most VPS have limits like that.  If it didn't kill it for using too much memory then it would kill it for CPU hogging shortly after.
